flask_jwt/__init__.py

```diff
@@ -71,53 +30,58 @@ def _default_response_handler(payload):
     'JWT_AUTH_USERNAME_KEY': 'username',
     'JWT_AUTH_PASSWORD_KEY': 'password',
     'JWT_ALGORITHM': 'HS256',
-    'JWT_VERIFY': True,
-    'JWT_VERIFY_EXPIRATION': True,
     'JWT_LEEWAY': 0,
-    'JWT_EXPIRATION_DELTA': timedelta(seconds=300),
     'JWT_AUTH_HEADER_PREFIX': 'JWT',
+    'JWT_EXPIRATION_DELTA': timedelta(seconds=300),
+    'JWT_VERIFY_CLAIMS': ['signature', 'exp', 'nbf', 'iat', 'aud'],
+    'JWT_REQUIRED_CLAIMS': ['exp', 'iat', 'nbf']
 }
 
 
-def jwt_required(realm=None):
-    """View decorator that requires a valid JWT token to be present in the request
+def _default_jwt_headers_handler(identity):
+    return None
 
-    :param realm: an optional realm
-    """
-    def wrapper(fn):
-        @wraps(fn)
-        def decorator(*args, **kwargs):
-            verify_jwt(realm)
-            return fn(*args, **kwargs)
-        return decorator
-    return wrapper
 
+def _default_jwt_payload_handler(identity):
+    return {
+        'exp': datetime.utcnow() + current_app.config.get('JWT_EXPIRATION_DELTA'),
+        'identity': getattr(identity, 'id') or identity['id']}
 
-class JWTError(Exception):
-    def __init__(self, error, description, status_code=400, headers=None):
-        self.error = error
-        self.description = description
-        self.status_code = status_code
-        self.headers = headers
 
+def _default_jwt_encode_handler(identity):
+    secret = current_app.config['JWT_SECRET_KEY']
+    algorithm = current_app.config['JWT_ALGORITHM']
+    payload = _jwt.jwt_payload_callback(identity)
+    headers = _jwt.jwt_headers_callback(identity)
+    return jwt.encode(payload, secret, algorithm=algorithm, headers=headers)
 
-def verify_jwt(realm=None):
-    """Does the actual work of verifying the JWT data in the current request.
-    This is done automatically for you by `jwt_required()` but you could call it manually.
-    Doing so would be useful in the context of optional JWT access in your APIs.
 
-    :param realm: an optional realm
-    """
-    realm = realm or current_app.config['JWT_DEFAULT_REALM']
-    auth = request.headers.get('Authorization', None)
+def _default_jwt_decode_handler(token):
+    secret = current_app.config['JWT_SECRET_KEY']
+    algorithm = current_app.config['JWT_ALGORITHM']
+
+    verify_claims = current_app.config['JWT_VERIFY_CLAIMS']
+    required_claims = current_app.config['JWT_REQUIRED_CLAIMS']
+
+    options = {
```

Comment:
Will be always equal to `None`.

Instead should look like

```
options = {
    'verify_{}'.format(claim): True
    for claim in verify_claims
}
options.update({
    'require_{}'.format(claim): True
    for claim in required_claims
})
```

Or somehow like this:

```
from itertools import chain

verify_claims = map(
    lambda x: ('verify_{}'.format(x), True),
    current_app.config['JWT_VERIFY_CLAIMS']
)
required_claims = map(
    lambda x: ('require_{}'.format(x), True),
    current_app.config['JWT_REQUIRED_CLAIMS']
)
options = dict(chain(verify_claims, required_claims))

```